server/odc-service/src/main/java/com/oceanbase/odc/service/monitor/MetricManager.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.service.monitor;
+
+import java.util.function.Supplier;
+
+import io.micrometer.core.instrument.Counter;
+
+public interface MetricManager {
+
+    void registerGauge(MeterKey meterKey, Supplier<Number> f);
```

Comment:
as we discussed before, Meter key is a inner concept, register api will return a object like a handler. user use this handler to interact with metricManager